version: '3.9'

services:
  ta-api:
    build:
      dockerfile: ./Dockerfile
    volumes:
      - ta-media-volume:/code_api/media
      - /tmp/app/mysqld:/run/mysqld
    ports:
      - 8051:8050
    restart: unless-stopped
    container_name: ta-api
    command: python manage.py runserver 0.0.0.0:8050
    env_file:
      - ./.env.development
    depends_on:
      - ta-database

  ta-database:
    container_name: ta-database
    image: mysql:8.0.33
    restart: always
    privileged: true
    environment:
      MYSQL_ROOT_PASSWORD: password
    volumes:
      - ./ta-mysql:/var/lib/mysql
    ports:
      - "3310:3306"
    command: --sql_mode="STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
  ta-phpmyadmin:
    container_name: ta-phpmyadmin
    image: phpmyadmin/phpmyadmin
    privileged: true
    ports:
      - '8101:80'
    restart: always
    environment:
      PMA_HOST: ta-database
      UPLOAD_LIMIT: 50M
    depends_on:
      - ta-database

volumes:
  ta-backend:
  ta-media-volume:
  ta-mysql:

